miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/QualityControlService.java

```diff
@@ -0,0 +1,34 @@
+package uk.ac.bbsrc.tgac.miso.service;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.stream.Collectors;
+
+import uk.ac.bbsrc.tgac.miso.core.data.QC;
+import uk.ac.bbsrc.tgac.miso.core.data.QualityControllable;
+import uk.ac.bbsrc.tgac.miso.core.data.type.QcType;
+
+public interface QualityControlService {
+
+  QC createQC(QC qc) throws IOException;
+
+  QC get(String target, Long id) throws IOException;
+
+  QualityControllable<?> getEntity(String target, long ownerId) throws IOException;
+
+  default QcType getQcType(String target, String name) throws IOException {
+    return listQcTypes().stream().filter(qcType -> qcType.getQcTarget().equals(target) && qcType.getName().equals(name)).findFirst()
+        .orElse(null);
+  }
+
+  Collection<? extends QC> listQCsFor(String target, long ownerId) throws IOException;
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Remove usage of generic wildcard type. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1452)